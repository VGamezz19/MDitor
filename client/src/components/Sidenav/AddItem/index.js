import React from 'react'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import AddItem from './AddItem'
import './AddItem.scss'

export default AddItem


/**
 * Exporting Cosmos Fixture
 */
function AddItemFixture() {
    return <MuiThemeProvider>
            <AddItem/>
        </MuiThemeProvider>
}

export { AddItemFixture }