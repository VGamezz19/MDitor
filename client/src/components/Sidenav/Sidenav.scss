@import '../../sass/_variables';
@import '../../sass/_extends';
.triggerIcon {
    @extend %medium-main-icon;
} 

.main-sidenav-content {
    background-color: $side-nav-main-color;
    .content-sidenav {
        @extend %flex-in-column;
        header {
            @extend %flex-in-column;
            align-items: center;
            height: 150px;
            padding: 1em;
            h4,
            h5 {
                text-align: center;
                color: $main-font-color
            }
            .delete-sidenav {
                align-self: flex-end;
                position: absolute;
                a {
                    padding: 0!important;
                    i {
                        @extend %medium-main-icon;
                        margin: 0!important;
                        font-size: 2rem !important;
                    }
                }
            }
        }
    }
}

//=== Sidenav ReactMaterialize
.side-nav a:hover {
    background-color: unset;
}

.collapsible {
    margin: 0 !important;
    border: 0 !important;
}