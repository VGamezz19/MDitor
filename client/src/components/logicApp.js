import File from 'mditor-types'

let appId = 123

const logicApp = {

    createFolder: (title, folders) => ([...folders, new File('folder', appId++, title)]),

    retrieveFolder: (id, folders) => {
        const folder = folders.find(folder => folder.id === id)
        return folder
    },

    updateFolder: (id, title, folders) => folders.map(folder => {
        if (folder.id === id) folder.setTitle(title)
        return folder
    }),

    removeFolder: (id, folders) => folders.filter(folder => folder.id !== id),

    createFile: (folderId, title, folders) => folders.map(folder => {
        if (folder.id === folderId) {
            folder.add(new File('file', appId++, title))
        }
        return folder
    }),

    retrieveFile: (folderId, id, folders) => {

        const folder = logicApp.retrieveFolder(folderId, folders)

        if (!folder) return false

        const file = folder.getFile(id)

        return file
    },

    removeFile: (folderId, id, folders) => folders.filter(folder => {
        if (folder.id === folderId) {
            folder.remove(folder.getFile(id))
        }
        return folder
    }),

    writeFile: (folderId, id, content, folders) => folders.map(folder => {
        if (folder.id === folderId) {
            folder.getFile(id).setContent(content)
        }
        return folder
    }),

    refactorDataToFileType: (dataFolder) => {
        let folders = []

        for (let i = 0; i < dataFolder.length; i++) {

            let filesToFolder = []
            const { id, title, files: folderFiles } = dataFolder[i];

            if (folderFiles) {

                let fileLength = folderFiles.length

                for (let i = 0; i < fileLength; i++) {

                    const { id, title, content } = folderFiles[i];

                    filesToFolder.push(new File('file', id, title, content))
                }
            }

            folders.push(new File("folder", id, title, undefined, filesToFolder))
        }
        return folders
    },

}

export default logicApp 