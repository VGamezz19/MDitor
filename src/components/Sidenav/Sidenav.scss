.black-font {
    color: black !important;
}

.content-sidenav {
    display: flex;
    flex-direction: column;
    header {
        align-items: center;
        height: 150px;
        align-content: center;
        padding: 1em;
        margin-top: 10px;
        display: flex;
        flex-direction: column;
        .delete-sidenav {
            align-self: flex-end;
            position: absolute;
            a {
                padding: 0!important;
                i {
                    margin: 0 !important;
                    font-size: 3em;
                    width: auto;
                }
            }
        }
        h4,
        h5 {
            color: white
        }
    }

    // Button add Folder
    button {
        align-self: center;
        margin-top: 2em;
    
    }
}

// === Collapsible Sidenav
.collapsible {
    .collapsible-header {
        display: flex;
        height: 70px;
        align-items: center;
        border-bottom: solid 1px grey;
        background-color: #9e9e9e  !important;
        i {
            margin-right: 2rem !important;
        }
    }
}

//=== Sidenav ReactMaterialize
.side-nav a:hover {
    background-color: unset;
    button {
        position: absolute;
    }
}