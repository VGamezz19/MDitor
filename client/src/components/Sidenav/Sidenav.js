import React, { Component } from 'react';
import { Fade } from 'react-reveal'
import PropTypes from 'prop-types';
import File from 'mditor-types';
import Drawer from 'material-ui/Drawer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { Link } from "react-router-dom"

import CircleButton from '../CircleButton'
import Folder from './Folder'
import AddItem from './AddItem'

class Sidenav extends Component {
    constructor(props) {
        super(props)

        this.state = {
            open: false,
            creatingFolder: false
        }
    }

    handleToggle = () => this.setState({ open: !this.state.open });

    handleClose = () => this.setState({ open: false });

    handlerCreatingFolder = () => {
        const { creatingFolder } = this.state

        if (creatingFolder) {

            return this.setState({ creatingFolder: false })
        }

        return this.setState({ creatingFolder: true })
    }

    onCreateFolder = (title) => {

        this.setState({ creatingFolder: false })

        this.props.logicFolder.create(title)
    }

    render() {
        const { buttonTriggerStyle, buttonDropStyle, user, folders, logicFolder } = this.props
        const { creatingFolder } = this.state

        return (
            <div>
                {this._renderTriggerButton(buttonTriggerStyle)}
                <MuiThemeProvider>
                    <Drawer
                        containerClassName='main-sidenav-content'
                        docked={false}
                        width={300}
                        open={this.state.open}
                        onRequestChange={(open) => this.setState({ open })}>
                        <section className='content-sidenav'>
                            {this._renderHeader(user, buttonDropStyle)}

                            {this._renderBodySidenav(folders, logicFolder)}

                            {this._renderCreateFolder(creatingFolder)}

                            <CircleButton className='add-folder grey' icon='add' onClick={this.handlerCreatingFolder} />
                        </section>
                    </Drawer>
                </MuiThemeProvider>
            </div>
        )
    }

    _renderTriggerButton({ className, icon }) {
        return (
            <CircleButton
                className={className}
                onClick={this.handleToggle}
                icon={icon} />
        )
    }

    _renderHeader({ name, surname }, { icon, className }) {
        if (name || surname) {
            return (
                <header>
                    <CircleButton
                        className={`delete-sidenav ${className}`}
                        onClick={this.handleToggle}
                        icon={icon} />
                    <Link to='/' onClick={this.handleToggle}>
                        <h4 id={'user-header'}>{name[0]}{surname[0]}</h4>
                        <h5>Welcome</h5>
                    </Link>

                </header>)
        }
        return false
    }

    _renderBodySidenav(folders, logicFolder) {
        return folders.map(folder => <Folder
            key={folder.getId()}
            logicFolder={logicFolder}
            folder={folder}
            handleToggleSidenav={this.handleToggle} />)
    }

    _renderCreateFolder(creatingFolder) {
        if (creatingFolder) {
            return (
                <Fade left>
                    <AddItem
                        itemType='folder'
                        inputType='text'
                        onSubmit={this.onCreateFolder}
                        onCancel={this.handlerCreatingFolder} />
                </Fade>
            )
        }
        return false
    }
}

Sidenav.propTypes = {
    /**
     * mandatory type
     * File from ('mditor-types');
     */
    folders: PropTypes.arrayOf(PropTypes.instanceOf(File)),
    /**
     * Object with logic
     */
    logiFolder: PropTypes.shape({
        create: PropTypes.func.isRequired,
        update: PropTypes.func.isRequired,
        delete: PropTypes.func.isRequired,
        logicFile: PropTypes.objectOf(PropTypes.func.isRequired)
    }),
    // logicFolder:  PropTypes.objectOf(PropTypes.func.isRequired),
}

export default Sidenav