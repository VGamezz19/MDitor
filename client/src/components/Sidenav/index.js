import Sidenav from './Sidenav'
import './Sidenav.scss'

export default Sidenav