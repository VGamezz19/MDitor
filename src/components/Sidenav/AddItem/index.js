import AddItem from './AddItem'
import './AddItem.scss'

export default AddItem