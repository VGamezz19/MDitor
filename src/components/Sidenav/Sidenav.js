import React, { Component } from 'react';
import { SideNav, SideNavItem, Button, Icon, Collapsible } from 'react-materialize'
import CircleButton from '../CircleButton'
import Folder from './Folder'
import { Fade } from 'react-reveal'
import AddItem from './AddItem'


//import { Button, Icon } from 'react-materialize'

class Sidenav extends Component {
    constructor(props) {
        super(props)
        this.state = {
            createFolder: false
        }
    }

    onClickCreateFolder = () => this.state.createFolder ? true : this.setState({ createFolder: true })

    handlerCreateForlder = (title) => {

        this.setState({ createFolder: false })

        this.props.setNewFolder(title)
    }

    render() {
        return (
            <SideNav
                className='side-nav grey lighten-2'
                trigger={
                    <Button
                        floating large
                        className={this.props.buttonTriggerStyle.buttonClassName}
                        waves='light'>
                        <Icon className='triggerIcon'> {this.props.buttonTriggerStyle.icon}</Icon>
                    </Button>
                }
                options={{
                    closeOnClick: true, // Closes side-nav on <a> clicks, useful for Angular/Meteor
                    onClose: function (el) { return console.log("Closing") }
                }}>

                <section className='content-sidenav'>
                    <header>
                        <SideNavItem className='delete-sidenav'>
                            <Icon className={this.props.buttonDropStyle.iconClassName}>{this.props.buttonDropStyle.icon}</Icon>
                        </SideNavItem>
                        <h4>{this.props.user.name[0]}{this.props.user.surname[0]}</h4>
                        <h5>Welcome</h5>
                    </header>

                    <Collapsible>
                        {this.props.folders.map(folder => <Folder 
                                                            onRenameFolder={this.props.onRenameFolder} 
                                                            title={folder.title} 
                                                            files={folder.files} />)}
                    </Collapsible>

                    {this.state.createFolder ?
                        <Fade left>
                            <AddItem
                                itemType='folder'
                                inputType='text'
                                onSubmit={this.handlerCreateForlder} />
                        </Fade>
                        : false}

                    <CircleButton ButtonClassName='add-folder grey' icon='add' onClick={this.onClickCreateFolder} />
                </section>

            </SideNav>
        )
    }
}

export default Sidenav



