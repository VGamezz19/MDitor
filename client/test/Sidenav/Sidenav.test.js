import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';

import Sidenav from '../../src/components/Sidenav';
import API from '../../src/api/ApiClient';
import logicApp from '../../src/components/logicApp';

const dataFolders = API.getFolders()

/**
 * Mandatory data type FILE
 */
const folders = logicApp.refactorDataToFileType(dataFolders)

const logicFolder = {
    create: () => console.log("Create Folder"),
    update: () => console.log("UpdateFolder"),
    delete: () => console.log("Delete Folder"),
    logicFile: {
        create: () => console.log("create File"),
        delete: () => console.log("delete File")
    }
}
const buttonTriggerStyle = { className: 'grey lighten-2', icon: 'menu' }
const buttonDropStyle = { className: 'grey lighten-2', icon: 'close' }
const user = { name: "Victor", surname: "Gamez" }

it('Main aplication rendering', () => {
    const div = document.createElement('div');

    const wrapper = <MemoryRouter><Sidenav
        folders={folders}
        logicFolder={logicFolder}
        buttonTriggerStyle={buttonTriggerStyle}
        buttonDropStyle={buttonDropStyle}
        user={user} />
    </MemoryRouter>

    ReactDOM.render(wrapper, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('Should create and Match same Snapshot', () => {

    const wrapper = <MemoryRouter><Sidenav
        folders={folders}
        logicFolder={logicFolder}
        buttonTriggerStyle={buttonTriggerStyle}
        buttonDropStyle={buttonDropStyle}
        user={user} />
    </MemoryRouter>

    const component = renderer.create(wrapper);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});