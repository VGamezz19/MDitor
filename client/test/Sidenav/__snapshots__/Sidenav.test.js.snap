// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create and Match same Snapshot 1`] = `
<div>
  <button
    className="btn waves-effect waves-light btn-large btn-floating main-icon grey lighten-2"
    disabled={false}
    icon="menu"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      menu
    </i>
  </button>
  <div
    className={undefined}
    style={undefined}
  >
    <div
      className={undefined}
      onClick={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0.54)",
          "height": "100%",
          "left": "-100%",
          "muiPrepared": true,
          "opacity": 0,
          "pointerEvents": "none",
          "position": "fixed",
          "top": 0,
          "transform": "translateZ(0)",
          "transition": "left 0ms cubic-bezier(0.23, 1, 0.32, 1) 400ms, opacity 400ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": "100%",
          "willChange": "opacity",
          "zIndex": 1200,
        }
      }
    />
    <div
      className="main-sidenav-content"
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": "0px",
          "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                 0 3px 10px rgba(0, 0, 0, 0.23)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "height": "100%",
          "left": 0,
          "muiPrepared": true,
          "overflow": "auto",
          "position": "fixed",
          "top": 0,
          "transform": "translate(-310px, 0)",
          "transition": "transform 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 300,
          "zIndex": 1300,
        }
      }
    >
      <section
        className="content-sidenav"
      >
        <header>
          <button
            className="btn waves-effect waves-light btn-large btn-floating main-icon delete-sidenav grey lighten-2"
            disabled={false}
            icon="close"
            onClick={[Function]}
          >
            <i
              className="material-icons"
            >
              close
            </i>
          </button>
          <h4
            id="user-header"
          >
            V
            G
          </h4>
          <h5>
            Welcome
          </h5>
        </header>
        <ul
          className="collapsible"
          data-collapsible="expandable"
        >
          <li
            onSelect={[Function]}
          >
            <div
              className="collapsible-header"
              onClick={[Function]}
            >
              <i
                className="material-icons"
              >
                chevron_right
                folder
              </i>
              new Folder (1)
            </div>
            <div
              className="icon-menu"
            >
              <div
                className={undefined}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                style={
                  Object {
                    "display": "inline-block",
                    "muiPrepared": true,
                    "position": "relative",
                  }
                }
              >
                <button
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOut={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                      "background": "none",
                      "border": 10,
                      "boxSizing": "border-box",
                      "cursor": "pointer",
                      "display": "inline-block",
                      "fontFamily": "Roboto, sans-serif",
                      "fontSize": 0,
                      "fontWeight": "inherit",
                      "height": 48,
                      "margin": 0,
                      "muiPrepared": true,
                      "outline": "none",
                      "overflow": "visible",
                      "padding": 12,
                      "position": "relative",
                      "textDecoration": "none",
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "verticalAlign": null,
                      "width": 48,
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <div
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                  >
                    <svg
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      style={
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "display": "inline-block",
                          "fill": "currentColor",
                          "height": 24,
                          "muiPrepared": true,
                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "userSelect": "none",
                          "width": 24,
                        }
                      }
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </div>
                </button>
                <div
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="collapsible-body grey lighten-2"
            >
              <ul
                className="content-files"
              >
                <li
                  className="li-item-file false"
                  onClick={[Function]}
                >
                  <a
                    className="files grey lighten-3\\""
                    href="/0/2/edit"
                    onClick={[Function]}
                  >
                    <button
                      className="files grey lighten-3"
                      type="submit"
                    >
                      new file (1)
                    </button>
                  </a>
                  <button
                    className="button-delete"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOut={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                        "background": "none",
                        "border": 10,
                        "boxSizing": "border-box",
                        "cursor": "pointer",
                        "display": "inline-block",
                        "fontFamily": "Roboto, sans-serif",
                        "fontSize": 0,
                        "fontWeight": "inherit",
                        "height": 48,
                        "margin": 0,
                        "muiPrepared": true,
                        "outline": "none",
                        "overflow": "visible",
                        "padding": 12,
                        "position": "relative",
                        "textDecoration": "none",
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "verticalAlign": null,
                        "width": 48,
                      }
                    }
                    tabIndex={0}
                    type="button"
                  >
                    <div
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                    >
                      <svg
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        style={
                          Object {
                            "color": "rgba(0, 0, 0, 0.87)",
                            "display": "inline-block",
                            "fill": "currentColor",
                            "height": 24,
                            "muiPrepared": true,
                            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                            "userSelect": "none",
                            "width": 24,
                          }
                        }
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                        />
                      </svg>
                    </div>
                  </button>
                </li>
                <li
                  className="li-item-file false"
                  onClick={[Function]}
                >
                  <a
                    className="files grey lighten-3\\""
                    href="/0/3/edit"
                    onClick={[Function]}
                  >
                    <button
                      className="files grey lighten-3"
                      type="submit"
                    >
                      new file (2)
                    </button>
                  </a>
                  <button
                    className="button-delete"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOut={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                        "background": "none",
                        "border": 10,
                        "boxSizing": "border-box",
                        "cursor": "pointer",
                        "display": "inline-block",
                        "fontFamily": "Roboto, sans-serif",
                        "fontSize": 0,
                        "fontWeight": "inherit",
                        "height": 48,
                        "margin": 0,
                        "muiPrepared": true,
                        "outline": "none",
                        "overflow": "visible",
                        "padding": 12,
                        "position": "relative",
                        "textDecoration": "none",
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "verticalAlign": null,
                        "width": 48,
                      }
                    }
                    tabIndex={0}
                    type="button"
                  >
                    <div
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                    >
                      <svg
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        style={
                          Object {
                            "color": "rgba(0, 0, 0, 0.87)",
                            "display": "inline-block",
                            "fill": "currentColor",
                            "height": 24,
                            "muiPrepared": true,
                            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                            "userSelect": "none",
                            "width": 24,
                          }
                        }
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                        />
                      </svg>
                    </div>
                  </button>
                </li>
                <button
                  className="btn waves-effect waves-light btn-large btn-floating main-icon add-files grey lighten-3"
                  disabled={false}
                  icon="add"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons"
                  >
                    add
                  </i>
                </button>
              </ul>
            </div>
          </li>
        </ul>
        <ul
          className="collapsible"
          data-collapsible="expandable"
        >
          <li
            onSelect={[Function]}
          >
            <div
              className="collapsible-header"
              onClick={[Function]}
            >
              <i
                className="material-icons"
              >
                chevron_right
                folder
              </i>
              new Folder (2)
            </div>
            <div
              className="icon-menu"
            >
              <div
                className={undefined}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                style={
                  Object {
                    "display": "inline-block",
                    "muiPrepared": true,
                    "position": "relative",
                  }
                }
              >
                <button
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOut={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                      "background": "none",
                      "border": 10,
                      "boxSizing": "border-box",
                      "cursor": "pointer",
                      "display": "inline-block",
                      "fontFamily": "Roboto, sans-serif",
                      "fontSize": 0,
                      "fontWeight": "inherit",
                      "height": 48,
                      "margin": 0,
                      "muiPrepared": true,
                      "outline": "none",
                      "overflow": "visible",
                      "padding": 12,
                      "position": "relative",
                      "textDecoration": "none",
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "verticalAlign": null,
                      "width": 48,
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <div
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                  >
                    <svg
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      style={
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "display": "inline-block",
                          "fill": "currentColor",
                          "height": 24,
                          "muiPrepared": true,
                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "userSelect": "none",
                          "width": 24,
                        }
                      }
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </div>
                </button>
                <div
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="collapsible-body grey lighten-2"
            >
              <ul
                className="content-files"
              >
                <li
                  className="li-item-file false"
                  onClick={[Function]}
                >
                  <a
                    className="files grey lighten-3\\""
                    href="/1/4/edit"
                    onClick={[Function]}
                  >
                    <button
                      className="files grey lighten-3"
                      type="submit"
                    >
                      new file (3)
                    </button>
                  </a>
                  <button
                    className="button-delete"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOut={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                        "background": "none",
                        "border": 10,
                        "boxSizing": "border-box",
                        "cursor": "pointer",
                        "display": "inline-block",
                        "fontFamily": "Roboto, sans-serif",
                        "fontSize": 0,
                        "fontWeight": "inherit",
                        "height": 48,
                        "margin": 0,
                        "muiPrepared": true,
                        "outline": "none",
                        "overflow": "visible",
                        "padding": 12,
                        "position": "relative",
                        "textDecoration": "none",
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "verticalAlign": null,
                        "width": 48,
                      }
                    }
                    tabIndex={0}
                    type="button"
                  >
                    <div
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                    >
                      <svg
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        style={
                          Object {
                            "color": "rgba(0, 0, 0, 0.87)",
                            "display": "inline-block",
                            "fill": "currentColor",
                            "height": 24,
                            "muiPrepared": true,
                            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                            "userSelect": "none",
                            "width": 24,
                          }
                        }
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                        />
                      </svg>
                    </div>
                  </button>
                </li>
                <li
                  className="li-item-file false"
                  onClick={[Function]}
                >
                  <a
                    className="files grey lighten-3\\""
                    href="/1/5/edit"
                    onClick={[Function]}
                  >
                    <button
                      className="files grey lighten-3"
                      type="submit"
                    >
                      new file (4)
                    </button>
                  </a>
                  <button
                    className="button-delete"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOut={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                        "background": "none",
                        "border": 10,
                        "boxSizing": "border-box",
                        "cursor": "pointer",
                        "display": "inline-block",
                        "fontFamily": "Roboto, sans-serif",
                        "fontSize": 0,
                        "fontWeight": "inherit",
                        "height": 48,
                        "margin": 0,
                        "muiPrepared": true,
                        "outline": "none",
                        "overflow": "visible",
                        "padding": 12,
                        "position": "relative",
                        "textDecoration": "none",
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "verticalAlign": null,
                        "width": 48,
                      }
                    }
                    tabIndex={0}
                    type="button"
                  >
                    <div
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                    >
                      <svg
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        style={
                          Object {
                            "color": "rgba(0, 0, 0, 0.87)",
                            "display": "inline-block",
                            "fill": "currentColor",
                            "height": 24,
                            "muiPrepared": true,
                            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                            "userSelect": "none",
                            "width": 24,
                          }
                        }
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                        />
                      </svg>
                    </div>
                  </button>
                </li>
                <button
                  className="btn waves-effect waves-light btn-large btn-floating main-icon add-files grey lighten-3"
                  disabled={false}
                  icon="add"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons"
                  >
                    add
                  </i>
                </button>
              </ul>
            </div>
          </li>
        </ul>
        <button
          className="btn waves-effect waves-light btn-large btn-floating main-icon add-folder grey"
          disabled={false}
          icon="add"
          onClick={[Function]}
        >
          <i
            className="material-icons"
          >
            add
          </i>
        </button>
      </section>
    </div>
  </div>
</div>
`;
